src/shared/view/hooks/usePagination.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useState, useCallback } from 'react';
+
+import { Pagination } from 'shared/view/components';
+
+const steps = [10, 25, 50, 100];
+
+export default function usePagination(items: any[]) {
```

Comment:
You need to handle `items.length` changing. We have `useOnChangeState` hook for this